front/app/containers/ProjectsShowPage/ProjectCTABar/index.tsx

```diff
@@ -57,20 +64,34 @@ const ProjectCTABar = ({ projectId }: ProjectCTABarProps) => {
     return () => window.removeEventListener('scroll', handleScroll);
   }, [isSmallerThanTablet]);
 
-  const participationMethod = project
-    ? getParticipationMethod(project.data, phases?.data)
-    : undefined;
-
-  if (!project || !participationMethod) {
+  if (!project || !phases) {
     return null;
   }
 
-  const BarContents = getMethodConfig(participationMethod).renderCTABar({
-    project: project.data,
-    phases: phases?.data,
-  });
-
-  if (isSticky && portalElement) {
+  const CTABars: { [method in ParticipationMethod]: JSX.Element | null } = {
+    ideation: <IdeationCTABar project={project.data} phases={phases.data} />,
+    proposals: <IdeationCTABar project={project.data} phases={phases.data} />,
+    native_survey: (
+      <NativeSurveyCTABar project={project.data} phases={phases.data} />
+    ),
+    information: <EventsCTABar project={project.data} phases={phases.data} />,
+    survey: (
+      <EmbeddedSurveyCTABar project={project.data} phases={phases.data} />
+    ),
+    voting: <VotingCTABar project={project.data} phases={phases.data} />,
+    poll: <PollCTABar project={project.data} phases={phases.data} />,
+    volunteering: (
+      <VolunteeringCTABar project={project.data} phases={phases.data} />
+    ),
+    document_annotation: (
+      <DocumentAnnotationCTABar project={project.data} phases={phases.data} />
+    ),
+  };
+
+  const participationMethod = getParticipationMethod(project.data, phases.data);
+  const CTABar = participationMethod ? CTABars[participationMethod] : null;
+
+  if (isSticky && CTABar && portalElement) {
```

Comment:
The CTABar fix should prevent the bug from happening again.

Reply:
I'll add a comment for this in the code.

Reply:
@amanda-anderson Left some comments to explain the brittle solution. I feel we better have tests, but that would require more refactoring so 🤞 the comments will hold this together.